src/main/scala/beam/agentsim/scheduler/BeamAgentScheduler.scala

```diff
@@ -207,55 +197,54 @@ class BeamAgentScheduler(val beamConfig: BeamConfig, stopTick: Double, val maxWi
 
     case Monitor =>
       log.debug(
-        s"\n\tnowInSeconds=$nowInSeconds,\n\tawaitingResponse.size=${awaitingResponse
-          .size()},\n\ttriggerQueue.size=${triggerQueue.size},\n\ttriggerQueue.head=${triggerQueue.headOption}\n\tawaitingResponse.head=$awaitingToString"
+        s"\n\tnowInSeconds=$nowInSeconds,\n\tawaitingResponse.size=${
+          awaitingResponse
+            .size()
+        },\n\ttriggerQueue.size=${triggerQueue.size},\n\ttriggerQueue.head=${triggerQueue.headOption}\n\tawaitingResponse.head=$awaitingToString"
       )
       awaitingResponse.values().forEach(x => log.debug("awaitingResponse:" + x.toString))
 
     case SkipOverBadActors =>
-      var numReps = 0L
-      currentTotalAwaitingResponse = awaitingResponse.values().stream().count()
-      if (currentTotalAwaitingResponse == previousTotalAwaitingRespone && currentTotalAwaitingResponse != 0) {
-        numberRepeats += 1
-        numReps = numberRepeats
-        log.debug("DEBUG: {} repeats.", numReps)
-      } else {
-        numberRepeats = 0
-      }
+      val stuckAgents = stuckFinder.detectStuckAgents()
```

Comment:
Should most of this be encapsulated into the detector itself? Then it could just be more like stuckfinder.register and unregister - letting it handle the check itself. Then you would only need to add the values to be monitored.

Reply:
I'd like to leave `StuckFinder` as simple as it is now. I think `BeamAgentScheduler` is right place to make decision what to do with stuck agents.

Reply:
In the sake of keeping the execution centralized, how about using a function then? The user of the class can inject their specific logic for what to do with stuck agents. I'm only pushing this one because it seems like a bit more back and forth - especially on removing and re-putting objects.